gk/main.c

```diff
@@ -1971,31 +1971,19 @@ gk_proc(void *arg)
 	struct rte_mbuf *front_icmp_bufs[gk_conf->front_max_pkt_burst];
 	struct rte_mbuf *back_icmp_bufs[gk_conf->back_max_pkt_burst];
 
-	int num_buckets = rte_hash_get_num_buckets(
-		instance->ip_flow_hash_table);
 	uint32_t bucket_idx = 0;
-	uint64_t last_scan_tsc = rte_rdtsc();
-	uint64_t last_measure_tsc = last_scan_tsc;
+	uint64_t last_measure_tsc = rte_rdtsc();
 	uint64_t basic_measurement_logging_cycles =
 		gk_conf->basic_measurement_logging_ms *
 		rte_get_tsc_hz() / 1000;
-	uint64_t bucket_scan_timeout_cycles = round(
-		(double)(gk_conf->flow_table_full_scan_ms *
-		rte_get_tsc_hz()) / (num_buckets * 1000.));
-	if (bucket_scan_timeout_cycles == 0) {
-		GK_LOG(WARNING,
-			"The value of the field flow_table_full_scan_ms in Gatekeeper configuration is too small\n");
-		exiting = true;
-		return -1;
-	}
+	uint32_t scan_iter = gk_conf->flow_table_scan_iter;
+	uint64_t iter_count = 0;
```

Comment:
`iter_count` is never bigger than `scan_iter`, so `iter_count` can be of type `uint32_t`.